eo-maven-plugin/src/main/java/org/eolang/maven/rust_project/Names.java

```diff
@@ -0,0 +1,135 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.rust_project;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.Locale;
+import java.util.concurrent.ConcurrentHashMap;
+import org.eolang.maven.footprint.FtDefault;
+import org.eolang.maven.util.Home;
+
+/**
+ * Uniquely converts the insert into the name for jni function.
+ * @since 0.30
+ */
+public final class Names {
+
+    /**
+     * Target directory.
+     * @checkstyle MemberNameCheck (7 lines)
+     */
+    private final Path dest;
+
+    /**
+     * All names.
+     */
+    private final ConcurrentHashMap<String, String> all;
+
+    /**
+     * Prefix for the name.
+     */
+    private final String prefix;
+
+    /**
+     * Ctor.
+     * @param target Directory where to serialize names.
+     * @throws IOException If any issues with IO.
+     */
+    @SuppressWarnings("PMD.ConstructorOnlyInitializesOrCallOtherConstructors")
+    public Names(final Path target) throws IOException {
+        this.dest = target.resolve("names");
+        this.prefix = target.toString().toLowerCase(Locale.ENGLISH).replaceAll("[^a-z0-9]", "x");
+        if (Files.exists(this.dest)) {
+            this.all = Names.load(this.dest);
+        } else {
+            this.all = new ConcurrentHashMap<>();
+        }
+    }
+
+    /**
+     * Assign the new name to the function.
+     * @param loc Location of the insert.
+     * @return The name.
+     */
+    public String name(final String loc) {
+        return this.all.computeIfAbsent(
+            loc,
+            key -> String.format(
+                "%s%d",
+                this.prefix,
+                this.all.size()
+            )
+        );
+    }
+
+    /**
+     * Saves the function to name dispatching table.
+     * @throws IOException If any issues with IO.
+     */
+    public void save() throws IOException {
+        Files.createDirectories(this.dest.getParent());
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        final ObjectOutputStream oos = new ObjectOutputStream(baos);
+        oos.writeObject(this.all);
+        oos.flush();
+        new Home(this.dest.getParent()).save(
+            new String(Base64.getEncoder().encode(baos.toByteArray())),
+            this.dest.getFileName()
+        );
+    }
+
+    /**
+     * Loads the table.
+     * @param src Path where to install from.
+     * @return The map.
+     * @throws IOException If any issues with IO.
+     */
+    private static ConcurrentHashMap<String, String> load(final Path src) throws IOException {
+        final String content = new FtDefault(src.getParent()).load(
```

Comment:
@levBagryansky Looks like you can inline that variable.

Reply:
@volodya-lombrozo if you talk about inserting its assignment to the place where it is used, It will be done by compiler if it is faster since the variable is final. At the same this code is clearer.

Reply:
@levBagryansky Please, take a look that article:
https://www.yegor256.com/2015/09/01/redundant-variables-are-evil.html

Reply:
@volodya-lombrozo ok, thanks!. Seems it is clearer for me only.